integreat_cms/static/src/js/analytics/hix-widget.ts

```diff
@@ -58,14 +62,71 @@ const setHixLabelState = (state: string) => {
         updateButton.classList.remove("hidden");
     }
 
+    // Hide HIX feedback when the HIX state is not up-to-date
+    if (state !== "updated") {
+        document.getElementById("hix-feedback").classList.add("hidden");
+    } else {
+        document.getElementById("hix-feedback").classList.remove("hidden");
+    }
+
     // Hide the loading spinner
     document.getElementById("hix-loading")?.classList.add("hidden");
 };
 
-const getHixValue = async () => {
+/**
+ * Mapping between category names in the HIX widget and in the Textlab API response
+ */
+const feedbackCategoriesMapping = new Map<string, string>([
+    ["hix-count-nested-sentences", "moreSentencesInClauses"],
+    ["hix-count-long-sentences", "moreSentencesInWords"],
+    ["hix-count-long-words", "moreWordsInLetters"],
+    ["hix-count-passive-voice-sentences", "countPassiveVoiceInSentence"],
+    ["hix-count-infinitive-constructions", "countInfinitiveConstructions"],
+    ["hix-count-nominal-sentences", "countNominalStyle"],
+    ["hix-count-future-tense-sentences", "countFutureTenseInSentence"],
+]);
+
+/**
+ * Display the HIX feedback details
+ */
+const updateHixFeedback = (hixFeedback: string) => {
+    const feedbackCategories: NodeListOf<HTMLElement> = document.querySelectorAll("[hix-feedback-category]");
+    const feedbackJson = JSON.parse(hixFeedback);
+
+    let feedbackCount = 0;
+
+    feedbackCategories.forEach((category) => {
+        const categoryName = category.getAttribute("hix-feedback-category");
+        const feedbackByCategory = feedbackJson[feedbackCategoriesMapping.get(categoryName)];
+
+        if (feedbackByCategory?.length > 0) {
+            category.classList.remove("hidden");
+            const categoryCount: HTMLElement = category.querySelector("span");
+            categoryCount.textContent = feedbackByCategory.length;
+            feedbackCount += 1;
+        } else {
+            category.classList.add("hidden");
+        }
+    });
+
+    const feedbackBlock: HTMLElement = document.getElementById("hix-feedback");
+    if (feedbackCount === 0) {
+        feedbackBlock.classList.add("hidden");
+    } else {
+        feedbackBlock.classList.remove("hidden");
+    }
```

Comment:
I'd suggest to filter the feedbackCategories at the start of this function and later on use the length of that array to make the if condition more clear.

#### Sample
```
    // Filter categories with feedback
    const categoriesWithFeedback = Array.from(feedbackCategories).filter((category) => {
        const categoryName = category.getAttribute("hix-feedback-category");
        const feedbackByCategory = feedbackJson[feedbackCategoriesMapping.get(categoryName)];
        return feedbackByCategory?.length > 0;
    });

    // Loop through each category
    categoriesWithFeedback.forEach((category) => {
        const categoryCount = category.querySelector("span");
        const categoryName = category.getAttribute("hix-feedback-category");
        const feedbackByCategory = feedbackJson[feedbackCategoriesMapping.get(categoryName)];
        categoryCount.textContent = feedbackByCategory.length.toString(); // Convert to string before setting
    });

    // Show/hide feedback block based on feedback count
    const feedbackBlock = document.getElementById("hix-feedback");
    if (categoriesWithFeedback.length === 0) {
        feedbackBlock.classList.add("hidden");
    } else {
        feedbackBlock.classList.remove("hidden");
    }
```

Reply:
const categoryName and const feedbackByCategory are created in 2 loops in this example

Reply:
Also missing is the behavior of hiding/displaying relevant feedback categories (divs) depending on categoryCount.
When count is 0 - the feedback category should be hidden in the widget, otherwise - it should be displayed.

Reply:
Sorry but it also doesn't work as expected.
Hiding/displaying relevant feedback categories (divs) depending on categoryCount is missing.
When count is 0 - the feedback category should be hidden in the widget, otherwise - it should be displayed.